### PR TITLE
[SxS 3] Add search side-by-side web page

### PR DESCRIPTION
**Pull request by joelverhagen**
Address https://github.com/NuGet/NuGetGallery/issues/7152.
Depends on https://github.com/NuGet/NuGetGallery/pull/7176.

1. Add .cshtml for the SxS page and associated CSS
1. Make minimally invasive changes to `_ListPackage.cshtml` to support SxS better
1. Add `ExperimentsController` to serve up the page and connect to the service layer

This is what it looks like:
image

### CODE REVIEW

File: src/NuGetGallery/Views/Experiments/SearchSideBySide.cshtml [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+                                    { "showEditButton", false },
+                                })
+                            newItemIndex++;
+                        }
+
+                        @if (!Model.NewSuccess)
+                        {
+                            <i>The new results could not be loaded.</i>
+                        }
+                        else if (Model.NewItems.Count == 0)
```
Comment (agr) [inline src/NuGetGallery/Views/Experiments/SearchSideBySide.cshtml]: `!Model.NewItems.Any()`?

**Comment by joelverhagen** [comment]
Spoke offline to @agr. This SxS is not meant as a replacement for blind evaluation of the new search results. That kind of blind, higher scale evaluation will be done with a randomized A/B test tracked here: https://github.com/NuGet/NuGetGallery/issues/7166.

A link to this web page will be handed out to internal/external customers directly or on Twitter so we can gather more English sentences about user experience. In other words, it's closer to a survey than an A/B test.

File: src/NuGetGallery/Views/Experiments/SearchSideBySide.cshtml [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                            Once you provide a search query, take a look at the results and fill out the form below so
+                            that we can understand your thoughts on how the new results can be improved. We're still
+                            playing with how the search query is parsed and how relevant packages are ranked so any
+                            input you can provide is helpful!
+                        </p>
+                        <p>
+                            If you'd like us to be able to reach out to your concerning some feedback you have left,
+                            please leave your email address. We'll read all of your feedback whether it is anonymous
+                            or not and would be happy to start a conversation over email if you have more ideas or
+                            concerns that don't fit well in the little comment box below.
```
Comment (agr) [inline src/NuGetGallery/Views/Experiments/SearchSideBySide.cshtml]: Do we need any privacy notice concerning that communication suggestion?

File: src/NuGetGallery/Views/Experiments/SearchSideBySide.cshtml (hunk 1 above)
Comment (joelverhagen) [reply src/NuGetGallery/Views/Experiments/SearchSideBySide.cshtml]: Sure

File: src/NuGetGallery/Views/Experiments/SearchSideBySide.cshtml (hunk 2 above)
Comment (joelverhagen) [reply src/NuGetGallery/Views/Experiments/SearchSideBySide.cshtml]: Great question. Let me sync with @karann-msft.

Comment (karann-msft) [reply src/NuGetGallery/Views/Experiments/SearchSideBySide.cshtml]: Your data will be respected and protected. Please see the [Microsoft Privacy Policy](https://privacy.microsoft.com/) for more information.

**Comment by joelverhagen** [comment]
> please do a quick accessibility pass using Keros

Done. It's all good but had to file https://github.com/microsoft/accessibility-insights-web/issues/692.

**Comment by scottbommarito** [comment]
@joelverhagen - thanks for filing that issue...that bug has been plaguing us for ages

**Pull request merged**